runtime/ts/storage/pouchdb/pouch-db-storage.ts

```diff
@@ -0,0 +1,245 @@
+// @
+// Copyright (c) 2017 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+import {assert} from '../../../../platform/assert-web.js';
+import {StorageBase, StorageProviderBase} from '../storage-provider-base.js';
+import {PouchDbKey} from './pouch-db-key.js';
+import {Id} from '../../id.js';
+import {Type} from '../../type.js';
+import {PouchDbCollection} from "./pouch-db-collection.js";
+import {PouchDbStorageProvider} from "./pouch-db-storage-provider.js";
+import {PouchDbBigCollection} from "./pouch-db-big-collection.js";
+import {PouchDbVariable} from "./pouch-db-variable.js";
+
+import PouchDB from 'pouchdb';
+import PouchDbMemory from 'pouchdb-adapter-memory';
+
+
+export class PouchDbStorage extends StorageBase {
+  // TODO(lindner) add global weak map of keys and handle replication events.
```

Comment:
Are we just being inefficient memory/perf wise by not doing this yet? Since I assume we handle replication events even with just this PR.

Reply:
It's a memory issue since all objects get retained in this map.  If javascript had weak references you'd use WeakValues for this map.  However it doesn't; so flagged to find a better solution or a way to garbage collect unreferenced objects.